From e075756548c349fa421112dc020a6f84f4a40d13 Mon Sep 17 00:00:00 2001
From: Rakesh Goyal <rgoyal@nvidia.com>
Date: Wed, 30 May 2012 16:35:21 +0530
Subject: [PATCH] set ro.commchip in init.c

set the ro.commchip for comms module present on platform using
odmdata.

bug 991513
bug 991610

Change-Id: I65ca20910fafd1a99c2c888c007d42922a61dd39
Reviewed-on: http://git-psac/r/222
Reviewed-by: Rakesh Goyal <rgoyal@nvidia.com>
Tested-by: Rakesh Goyal <rgoyal@nvidia.com>
Reviewed-by: Varun Colbert <vcolbert@nvidia.com>
---
 init/init.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/init/init.c b/init/init.c
index 4f57144..7648d10 100755
--- a/init/init.c
+++ b/init/init.c
@@ -72,6 +72,7 @@ static int   bootchart_count;
 static char console[32];
 static char bootmode[32];
 static char hardware[32];
+static char commchip[32];
 static unsigned revision = 0;
 static char qemu[32];
 
@@ -632,6 +633,13 @@ static void import_kernel_nv(char *name, int for_emulator)
         cnt = snprintf(prop, sizeof(prop), "ro.boot.%s", boot_prop_name);
         if (cnt < PROP_NAME_MAX)
             property_set(prop, value);
+    } else if (!strncmp(name,"commchip_id",11)){
+        char prop[PROP_NAME_MAX];
+        int cnt;
+        cnt = snprintf(prop, sizeof(prop), "ro.boot.%s", name);
+        if (cnt < PROP_NAME_MAX)
+            property_set(prop, value);
+
     }
 }
 
@@ -649,6 +657,7 @@ static void export_kernel_boot_props(void)
         { "ro.boot.mode", "ro.bootmode", "unknown", },
         { "ro.boot.baseband", "ro.baseband", "unknown", },
         { "ro.boot.bootloader", "ro.bootloader", "unknown", },
+        { "ro.boot.commchip_id", "ro.commchip", "unknown", },
     };
 
     for (i = 0; i < ARRAY_SIZE(prop_map); i++) {
@@ -680,6 +689,11 @@ static void export_kernel_boot_props(void)
         property_set("ro.factorytest", "2");
     else
         property_set("ro.factorytest", "0");
+
+    pval = property_get("ro.boot.commchip_id");
+    if (pval)
+        strlcpy(commchip, pval, sizeof(console));
+    property_set("ro.commchip", commchip[0] ? commchip : "unknown");
 }
 
 static void process_kernel_cmdline(void)
-- 
1.7.0.4

